src/analysis/retail/shaman/restoration/modules/talents/EarthenHarmony.tsx

```diff
@@ -0,0 +1,129 @@
+import SPELLS from 'common/SPELLS';
+import talents from 'common/TALENTS/shaman';
+import { WCLDamageTaken, WCLDamageTakenTableResponse } from 'common/WCL_TYPES';
+import fetchWcl from 'common/fetchWclApi';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import { calculateEffectiveHealing } from 'parser/core/EventCalculateLib';
+import Events, { EventType, HealEvent } from 'parser/core/Events';
+import Combatants from 'parser/shared/modules/Combatants';
+
+const DAMAGE_REDUCTION_PER_POINT = 0.03;
+const HEALING_INCREASE_PER_POINT = 0.5;
+const HEALTH_THRESHOLD = 0.75;
+
+class EarthenHarmony extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+  };
+
+  protected combatants!: Combatants;
+  damageReduction;
+  damageTakenWithEarthShield: number = 0;
+  damageTakenWithElementalOrbitEarthShield: number = 0;
+  healingIncrease;
+  earthShieldHealing: number = 0;
+  elementalOrbitEarthShieldHealing: number = 0;
+  elementalOrbitActive: boolean = false;
+  constructor(options: Options) {
+    super(options);
+    this.active = this.selectedCombatant.hasTalent(talents.EARTHEN_HARMONY_TALENT);
+    this.elementalOrbitActive = this.selectedCombatant.hasTalent(talents.ELEMENTAL_ORBIT_TALENT);
+    this.damageReduction =
+      DAMAGE_REDUCTION_PER_POINT *
+      this.selectedCombatant.getTalentRank(talents.EARTHEN_HARMONY_TALENT);
+    this.healingIncrease =
+      HEALING_INCREASE_PER_POINT *
+      this.selectedCombatant.getTalentRank(talents.EARTHEN_HARMONY_TALENT);
+
+    if (!this.active) {
+      return;
+    }
+    this.addEventListener(
+      Events.heal.by(SELECTED_PLAYER).spell(SPELLS.EARTH_SHIELD_HEAL),
+      this.onEarthShieldHeal,
+    );
+    this.loadDamageTakenDuringEarthShield();
+    if (this.elementalOrbitActive) {
+      this.loadDamageTakenDuringEarthShieldElementalOrbit();
+    }
+  }
+
+  get totalHealing() {
+    return this.earthShieldHealing + this.elementalOrbitEarthShieldHealing;
+  }
+
+  get earthShielddamageReduced() {
+    return (this.damageTakenWithEarthShield / (1 - this.damageReduction)) * this.damageReduction;
+  }
+
+  get elementalOrbitDamageReduced() {
+    return (
+      (this.damageTakenWithElementalOrbitEarthShield / (1 - this.damageReduction)) *
+      this.damageReduction
+    );
+  }
+
+  onEarthShieldHeal(event: HealEvent) {
+    if (!this.targetIsBelowHpThreshold(event)) {
+      return;
+    }
+    const combatant = this.combatants.getEntity(event);
```

Comment:
Would reduce redundant checks if you did 
```
if (!combatant) {
	return;
}
```

Reply:
good point